core/dom/range.js

```diff
@@ -1261,6 +1261,54 @@ CKEDITOR.dom.range = function( root ) {
 		enlarge: function( unit, excludeBrs ) {
 			var leadingWhitespaceRegex = new RegExp( /[^\s\ufeff]/ );
 
+			// Function check if there are only whitespaces from the given starting point
+			// (startContainer and startOffset) till the end on block.
+			// Examples ("[" is the start point):
+			//  - <p>foo[ </p>           - will return true,
+			//  - <p><b>foo[ </b> </p>   - will return true,
+			//  - <p>foo[ bar</p>        - will return false,
+			//  - <p><b>foo[ </b>bar</p> - will return false,
+			//  - <p>foo[ <b></b></p>    - will return false.
+			function onlyWhiteSpaces( startContainer, startOffset, boundary, siblingText ) {
+				// We need to enlarge range if there is white space at the end of the block,
+				// because it is not displayed in WYSIWYG mode and user can not select it. So
+				// "<p>foo[bar] </p>" should be changed to "<p>foo[bar ]</p>". On the other hand
+				// we should do nothing if we are not at the end of the block, so this should not
+				// be changed: "<p><i>[foo] </i>bar</p>".
+				var walkerRange = new CKEDITOR.dom.range( boundary );
+				walkerRange.setStart( startContainer, startOffset );
+				// The guard will find the end of range so I put boundary here.
+				walkerRange.setEndAt( boundary, CKEDITOR.POSITION_BEFORE_END );
+
+				var walker = new CKEDITOR.dom.walker( walkerRange ),
+					node;
+
+				walker.guard = function( node ) {
+					// Stop if you exit block.
+					return !( node.type == CKEDITOR.NODE_ELEMENT && node.isBlockBoundary() );
+				};
+
+				while ( ( node = walker.next() ) ) {
+					if ( node.type != CKEDITOR.NODE_TEXT ) {
+						// Stop if you enter to any node (walker.next() will return node only
+						// it goes out, not if it is go into node).
+						return false;
+					} else {
+						// Trim the first node to startOffset.
+						if ( node != startContainer )
+							siblingText = node.getText();
```

Comment:
The `siblingText` variable is overridden here but it's also used outside of the function. Isn't there a case when not overriding the variable outside of this function would generate an incorrect result (like in line 1577)?

Reply:
TBH - I just put my trust in all tests we have so far... Also, those are the most complicated ones. It is even hard to name them correctly - since there are so many `if` cases to handling the `enlarging` the range 🤔 Eventually we should decide to create a separate PR for writing separate tests and refactor range functions. IMHO - since those logics are very old - we could say that they work stable. I'm not fully convinced whether we should make additional changes here without any explicit issue?

Reply:
I'm not able to come up with a case that would break it 🤔 And you're right, our test case for ranges is pretty extensive and should catch it anyway.